### PR TITLE
Add fallback for missing target in Scrollspy

### PR DESCRIPTION
**Pull request by Chaostheorie**
Fix  #33943 by adding the suggested check by @yura-x with fallback to console warning. The current fallback message can most likely be enhanced, but it was the best/ most concise thing I could come up with for now.

### CODE REVIEW

File: js/src/scrollspy.js [hunk 1]
```diff
@@ -221,30 +221,35 @@ class ScrollSpy extends BaseComponent {
 
     const link = SelectorEngine.findOne(queries.join(','), this._config.target)
 
-    link.classList.add(CLASS_NAME_ACTIVE)
-    if (link.classList.contains(CLASS_NAME_DROPDOWN_ITEM)) {
-      SelectorEngine.findOne(SELECTOR_DROPDOWN_TOGGLE, link.closest(SELECTOR_DROPDOWN))
-        .classList.add(CLASS_NAME_ACTIVE)
+    if (link !== undefined && link !== null) {
```
Comment (XhmikosR) [inline js/src/scrollspy.js:224]:
```js
    if (link) {
```

should be enough, no? Just make sure you swap the conditions if you go with this approach.

**Comment by XhmikosR** [comment]
@alpa @GeoSot what's your take on this?

File: js/src/scrollspy.js (hunk 1 above)
Comment (Chaostheorie) [reply js/src/scrollspy.js:224]: Thank you for the comment. That's better indeed. Will make the change asap

Comment (XhmikosR) [reply js/src/scrollspy.js:224]: Or actually, just do `if (!link)`. But this needs tests for sure either way.

**Comment by Chaostheorie** [comment]
Uhm, the linting settings are configured to prohibit the usage of console. Is there a recommended way for sending warnings or are warnings not intended, and it should throw an error instead?

**Pull request closed without merging**